rules/gather_licenses_info.bzl

```diff
@@ -11,21 +11,22 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
 """Rules and macros for collecting LicenseInfo providers."""
 
+load(
+    "@rules_license//rules:licenses_core.bzl",
+    "TraceInfo",
+    "gather_licenses_info_common",
+    "should_traverse",
+)
 load(
     "@rules_license//rules:providers.bzl",
-    "LicenseInfo",
-    "LicensesInfo",
+    "TransitiveLicensesInfo",
 )
 
-# Debugging verbosity
-_VERBOSITY = 0
-
-def _debug(loglevel, msg):
-    if _VERBOSITY > loglevel:
-        print(msg)  # buildifier: disable=print
+# Definition for compliance namespace, used for filtering licenses
+# based on the namespace to which they belong.
+NAMESPACES = ["compliance"]
```

Comment:
@wcn3 @aiuto this definition of `NAMESPACES` and its use in `rules/gather_licneses_info.bzl` line 144 (also in this PR) causes `gather_metadata_info_common` (the implementation for `gather_licenses_info` aspect in `rules/gather_licenses_info.bzl`) to only pick up `LicenseInfo` providers with `namespace = 'compliance'`. One effect this has is policy checks implemented using `gather_licenses_info` will not fail when they should because they ignore licenses attached to dependencies that do not have `namespace = compliance` during traversal of dependencies.

The `license` rule definition contains:
```
        "namespace": attr.string(
            doc = "A human readable name used to organize licenses into categories." +
                  " This is used in google3 to differentiate third party licenses used" +
                  " for compliance versus internal licenses used by SLAsan for internal" +
                  " teams' SLAs.",
        ),
```
I see this is a useful attr, but it should be clear to users that a `license` that does not have `namespace = compliance` will act different. Additionally, this doc seems to contain google-internal information that can be removed.

Could you please clarify this logic of treating "compliance" namespace differently? Wondering if it should be removed entirely due to its effect on `gather_licenses_info` or if I am missing something.

Reply:
I think it can be removed.